app/services/form1010cg/service.rb

```diff
@@ -3,25 +3,85 @@
 # This service manages the interactions between CaregiversAssistanceClaim, CARMA, and Form1010cg::Submission.
 module Form1010cg
   class Service
-    def submit_claim!(claim_data)
-      claim = SavedClaim::CaregiversAssistanceClaim.new(claim_data)
+    attr_reader :claim
+
+    NOT_FOUND = 'NOT_FOUND'
+
+    def initialize(claim)
       claim.valid? || raise(Common::Exceptions::ValidationErrors, claim)
 
-      carma_submission = CARMA::Models::Submission.from_claim(claim)
+      @claim = claim
+      @cached_icns = {}
+    end
 
-      carma_submission.submit!
+    def process_claim!
+      assert_veteran_status
+
+      carma_submission = CARMA::Models::Submission.from_claim(claim, build_metadata).submit!
 
       Form1010cg::Submission.new(
         carma_case_id: carma_submission.carma_case_id,
         submitted_at: carma_submission.submitted_at
       )
     end
 
+    def assert_veteran_status
+      raise_unprocessable if icn_for('veteran') == NOT_FOUND
+    end
+
+    def build_metadata
+      claim.form_subjects.each_with_object({}) do |form_subject, metadata|
+        icn = icn_for(form_subject)
+        metadata[form_subject.to_sym] = {
+          icn: icn == NOT_FOUND ? nil : icn
+        }
+      end
+    end
+
+    def icn_for(form_subject)
+      cached_icn = @cached_icns[form_subject]
+      return cached_icn unless cached_icn.nil?
+
+      begin
+        response = mvi_service.find_profile(build_user_identity_for(form_subject))
+      rescue MVI::Errors::RecordNotFound
+        return @cached_icns[form_subject] = NOT_FOUND
+      end
+
+      @cached_icns[form_subject] = response&.profile&.icn if response&.status == 'OK'
+    end
+
     private
 
-    # Destroy this form it has previously been stored in-progress by this user_context
-    def form_schema_id
-      SavedClaim::CaregiversAssistanceClaim::FORM
+    def raise_unprocessable
+      message = 'Unable to process submission digitally'
+      claim.errors.add(:base, message, message: message)
+      raise(Common::Exceptions::ValidationErrors, claim)
+    end
+
+    def mvi_service
+      @mvi_service ||= MVI::Service.new
+    end
+
+    def build_user_identity_for(form_subject)
+      data = claim.parsed_form[form_subject]
+
+      attributes = {
+        first_name: data['fullName']['first'],
+        middle_name: data['fullName']['middle'],
+        last_name: data['fullName']['last'],
+        birth_date: data['dateOfBirth'],
+        gender: data['gender'] == 'U' ? nil : data['gender'],
+        ssn: data['ssnOrTin'],
+        email: data['email'] || 'no-email@example.com',
+        uuid: SecureRandom.uuid,
```

Comment:
What are the consequences of sending a uuid to MVI that is not a user's uuid. Might MVI save this uuid and corrupt the user's record?

Reply:
@annaswims The uuid never gets passed to the API request to MVI. I'm building a `UserIdentity` since `MVI::Service#find_profile` requires a valid `UserIdentity` and these are the requirements of a valid user identity.

The request to MVI will only send `first_name`, `middle_name`, `last_name`, `birth_date`, `ssn`, and `gender`.
```rb
    # lib/mvi/service.rb:198
    def message_user_attributes(user)
      given_names = [user.first_name]
      given_names.push user.middle_name unless user.middle_name.nil?
      profile = {
        given_names: given_names,
        last_name: user.last_name,
        birth_date: user.birth_date,
        ssn: user.ssn,
        gender: user.gender
      }
      MVI::Messages::FindProfileMessage.new(profile).to_xml
    end
```

A better alternative would be to create an `MVI::Service#find_profile_with_attributes` that would only require `first_name`, `middle_name`, `last_name`, `birth_date`, `ssn`, and `gender` vs and entire `UserIdentity`. This would avoid confusion and (more importantly) eliminate the risk of a future developer using this `UserIdentity` for other purposes (like a `put` or `patch`).

Thoughts?

Reply:
HCA also has code that queries MVI to look up a user's ICN based on traits.
Check out [`MVI::AttrService`](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/mvi/attr_service.rb).
HCA [feeds it ](https://github.com/department-of-veterans-affairs/vets-api/blob/f6fcced922e40034eb4fb230c2f65a8876bb49c0/app/controllers/v0/health_care_applications_controller.rb#L37) an [`HCA::UserAttribute` object](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/hca/user_attributes.rb).
The `HCA::UserAttribute` object exists for this exact purpose: serves an object that "duck types" with the necessary attributes to facilitate an MVI traits search.
It was created in the `HCA` namespace because that was the first use for it.
Now that we have multiple use cases for it, we should probably move it to another namespace (i.e. `MVI`) and re-use it here so we keep things "DRY", and hopefully reduce the likelihood of running into confusing scenarios like this again.

Reply:
Looks like the `MVI` lib already has a model for requiring the proper attributes in a search: `MVI::Messages::FindProfileMessage` (in _/lib/mvi/messages/find_profile_message.rb_). It even contains the validation required.

@jholton, @annaswims, What are your thoughts on this refactor:

Implement a `MVI::Service#find_profile` and a `MVI::Service#find_profile_by_user_identity`. 

- `#find_profile` requires a valid `MVI::Messages::FindProfileMessage`, `MVI::Messages::FindProfileMessageIcn`, or `MVI::Messages::FindProfileMessageEdipi`. These models already exist. Any consumers of the service is responsible for building a valid message for the `#find_profile` call.

- `#find_profile_by_user_identity` would require a valid `UserIdentity` and converts it to a valid `MVI::Messages::FindProfileMessage` | `MVI::Messages::FindProfileMessageIcn` | `MVI::Messages::FindProfileMessageEdipi`, then calls the new `#find_profile` method.

Reply:
I will do a followup PR to make this change. Here's the ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/9083

Reply:
@jholton I was working on this change and realized that the `MVI::AttrService` is tightly coupled with HCA, and having CARMA use it wouldn't make sense. It has it's own Faraday client and configuration that specifies HCA (https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/mvi/attr_configuration.rb).

I'm wondering if I should (1) merge the `MVI::AttrService`'s features into `MVI::Service` so all other consumers can use the same generic client, or (2) follow suit and make a CARMA MVI service with its own Faraday client and config.

I'm leaning towards Number 1 but I don't what the reason reasons for `MVI::AttrService` having its own client other than having that service name "**HCAMVI**" show up in logs and metrics separately from "**MVI**". Is this the behavior we want?